src/wkt_to_projjson.jl

```diff
@@ -0,0 +1,1773 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+module SafeIterate
+export safe_iterate
+function safe_iterate(iterator, optional_state::Union{Tuple{},Tuple{Any}})
+  if optional_state === ()
+    iterate(iterator)
+  else
+    iterate(iterator, only(optional_state))
+  end
+end
+end
+
+module ParseTrees
```

Comment:
From a quick look into the PR, it seems that you are implementing a parser from scratch. I wonder if we could take a more direct route with a new dependency? For example, I know Automa.jl provides methods to define performant parsers from regular expressions and auxiliary functions: https://biojulia.dev/Automa.jl/stable/parser

@eliascarv do you have that Automa.jl experiment with you somewhere? It would be nice to share it in https://github.com/JuliaEarth/GeoIO.jl/issues/150

Reply:
> I wonder if we could take a more direct route with a new dependency?

For the record, a lot of the code in this PR is very generic and suitable to be its own package, if you find that acceptable I could split it out into a package (or several packages) and then depend on the new package in this PR?

Reply:
NB: depending on Automa and depending on these new packages are independent considerations, the new packages would not concern lexing.

Reply:
> For the record, a lot of the code in this PR is very generic and suitable to be its own package, if you find that acceptable I could split it out into a package (or several packages) and then depend on the new package in this PR?

What would be the scope of this new package? A WKT2Parser.jl? Or a more generic package for parsing formats of a certain kind?

Reply:
> What would be the scope of this new package?

The `ParseTrees` module is independent of any specific language, such as JSON or WKT, so I'll make it one package, so the specific formats could depend on it. Then there would also be:
* a package for (un)parsing JSON, depending on ParseTrees.jl
* a package for (un)parsing WKT, depending on ParseTrees.jl

Come to think of it I'll be doing that right now in any case, as it seems like it might be useful to the community. While several packages for parsing and printing JSON already exist, none have the homogeneous/type-stable design of TypeTrees, as far as I'm aware.

Reply:
Alternatively, perhaps I should start by creating one monolithic package that supports both JSON, WKT, and perhaps some other formats, then later split out other subpackages.

Reply:
Seems like a very nice contribution. Looking forward to it.